sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/ScannerForMSBuildInstaller.java

```diff
@@ -195,16 +202,16 @@ private static File locateInstalledScript(@Nullable Version scannerVersion, File
         filename = "SonarQube.Scanner.MSBuild.exe";
       }
     }
-    File script = new File(toDir, directoryName(scannerVersion) + "/" + filename);
+    File script = new File(toDir, directoryName(scannerVersion, useDotNetCore) + File.separator + filename);
     if (!script.exists()) {
       throw new IllegalStateException("File does not exist: " + script);
     }
     return script;
   }
 
-  private static String directoryName(@Nullable Version scannerVersion) {
+  private static String directoryName(@Nullable Version scannerVersion, boolean useDotNetCore) {
     if (scannerVersion != null) {
-      return "sonar-scanner-msbuild-" + scannerVersion.toString();
+      return "sonar-scanner-msbuild-" + scannerVersion.toString()+"-"+getClassifier(useDotNetCore);
```

Comment:
[minor][cosmetic] You are missing some spaces around `+`. I'll let you see if you want to fix this or not.